flang/include/flang/ISO_Fortran_binding.h

```diff
@@ -32,7 +32,9 @@
 /* 18.5.4 */
 #define CFI_VERSION 20240719
 
+#ifndef CFI_MAX_RANK
```

Comment:
Since this is a user-facing interface, we don't want to make it too easy for users to try to subvert this definition for their own programs.  I suggest that you also require that a particular macro name also be defined in order to perform this override; perhaps something like

```
#if !defined CFI_MAX_RANK || !defined __OVERRIDE_CFI_MAX_RANK
...
```

That way, a naive user wanting to change CFI_MAX_RANK by simply defining it to (say) a larger number will be discouraged